### ISSUE TITLE
RAVEN Build Error on Windows

### ISSUE THREAD

**Issue by KwesiBoafo** [idaholab/raven#1241]
--------
Defect Description
--------
I tried to install the windows version of RAVEN and upon attempting to build raven, I get the following error message:
"error: Error executing cmd /u /c "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64 && set"

I have attempted several ways to get it to work but to no avail. Kindly provide guidance on how to address this error.

**Describe the defect**

##### What did you expect to see happen?
For RAVEN to build successfully

##### What did you see instead?

An error message as described above.

##### Do you have a suggested fix for the development team?
None

**Describe how to Reproduce**
1. 
2. 
3. 
4. 

**Screenshots and Input Files**
Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.
![image](https://user-images.githubusercontent.com/65812661/82729534-d5689200-9ce7-11ea-9c2a-2eaeef88abe8.png)

**Platform (please complete the following information):**
 - OS: [e.g. iOS] Windows
 - Version: [e.g. 22]
 - Dependencies Installation: [CONDA or PIP] Conda

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

**Comment by tyler477** [comment]
I have a similar issue, currently. Did you ever end up fixing the issue? If so, could you kindly give some insight? Much appreciated!

**Comment by alfoa** [comment]
Hi @tyler477 and @KwesiBoafo . It looks like that we missed this issue and we deeply apologize.

Are you still experiencing this problem? If so (@tyler477  since you just commented on this issue), can you attach your error?
In addition, are you using the CMD or the GitBash enviroment? If you are using the CMD, can you try in GitBash directly?
Thank you

**Comment by tyler477** [comment]
@alfoa, Thanks so so much for a quick response!

Yes, I am still experiencing the issue (attached screenshot below). I am using the CMD on windows 10. I will try momentarily on GitBash directly. 

![CMD](https://user-images.githubusercontent.com/70550749/97246645-33882c80-17d4-11eb-8946-b817eba95694.PNG)

**Comment by alfoa** [comment]
Okay. Thank you. Let me check in my remote win machine.
In the meanwhile, (sorry in advance for the silly question). Did the Visual Studio installation go through (at least "Desktop development with C++")
Andrea

**Comment by tyler477** [comment]
Yes. I have Visual Studio installed and working on my desktop with the C++ extension 

Tyler

**Comment by KwesiBoafo** [comment]
Hi Andrea, 

I couldn't resolve the issues with raven and decided to switch to a
different software a while back. I might let a student try out your
suggestions in their research and will let you know how it goes. By the
way, what is the status of the raven interface with MELCOR code? which I
understand was under development? 

Best regards, 

Emmanuel 

On 2020-10-27 01:36, Andrea Alfonsi - INL wrote:

> Hi @tyler477 [1] and @KwesiBoafo [2] . It looks like that we missed this issue and we deeply apologize. 
> 
> Are you still experiencing this problem? If so (@tyler477 [1] since you just commented on this issue), can you attach your error?
> In addition, are you using the CMD or the GitBash enviroment? If you are using the CMD, can you try in GitBash directly?
> Thank you 
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [3], or unsubscribe [4].

 

Links:
------
[1] https://github.com/tyler477
[2] https://github.com/KwesiBoafo
[3] https://github.com/idaholab/raven/issues/1241#issuecomment-716920530
[4]

**Comment by alfoa** [comment]
@tyler477 can I ask you to print and paste your path here (in GitBash):

``` echo $PATH```

**Comment by tyler477** [comment]
@alfoa I am using visual studio version 1.43.2

After printing my path, it returned: 

/c/Users/Tyler/bin:/mingw64/bin:/usr/local/bin:/usr/bin:/bin:/mingw64/bin:/usr/bin:/c/Users/Tyler/bin:/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/c/Program Files (x86)/Common Files/Oracle/Java/javapath:/c/WINDOWS/system32:/c/WINDOWS:/c/WINDOWS/System32/Wbem:/c/WINDOWS/System32/WindowsPowerShell/v1.0:/c/WINDOWS/System32/OpenSSH:/c/Program Files/MATLAB/R2020b/runtime/win64:/c/Program Files/MATLAB/R2020b/bin:/c/Program Files/MATLAB/R2018b/bin:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/110/Tools/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/Tools/Binn:/c/Program Files/Microsoft SQL Server/120/Tools/Binn:/c/Program Files/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Windows Kits/8.1/Windows Performance Toolkit:/cmd:/mingw64/bin:/usr/bin:/c/Program Files/PuTTY:/c/Users/Tyler/miniconda3:/c/Users/Tyler/miniconda3/Library/mingw-w64/bin:/c/Users/Tyler/miniconda3/Library/usr/bin:/c/Users/Tyler/miniconda3/Library/bin:/c/Users/Tyler/miniconda3/Scripts:/c/Users/Tyler/AppData/Local/Microsoft/WindowsApps:/c/Users/Tyler/AppData/Local/Programs/Microsoft VS Code/bin:/usr/bin/vendor_perl:/usr/bin/core_perl

Is this what you are looking for?

Thanks, Tyler.

**Comment by tyler477** [comment]
I have just updated Visual Studio to the latest September 2020 version. But I still have received
 this same error when trying "bash.exe build_raven", from both CMD and GitBash.

**Comment by alfoa** [comment]
The Path is correct (Visual Studio is there). @joshua-cogliati-inl I need your expertise here.

**Comment by joshua-cogliati-inl** [comment]
Hm, that used to mean that VS90COMNTOOLS is not set, but I thought that newer versions of visual studio did not need it set.

**Comment by tyler477** [comment]
hmm... I will keep trying to troubleshoot this. Thanks for your support @alfoa  and @joshua-cogliati-inl!

**Comment by joshua-cogliati-inl** [comment]
Also, usually we run in the git bash environment, instead of just running bash.exe when we need it.  That might make a difference.

**Comment by alfoa** [comment]
@tyler477 mybe an option can be to follow the instructions at https://github.com/idaholab/raven/wiki/installationWindowsOld#2-let-the-build-system-know-where-to-find-the-compiler 

(it should not be required but you can give a shoot and let us know maybe? )
Sorry for this.

**Comment by alfoa** [comment]
@tyler477 I am going to try to re-install the code with the latest versions of visual studio (starting from scratch) to see if I can replicate the problem.

**Comment by tyler477** [comment]
@alfoa I will try this thank you! So sorry for the trouble.

**Comment by tyler477** [comment]
@alfoa Thank you so much for your help! I think I've finally got it working... the issue appeared to be that I had the wrong version of Visual Studio installed. Before I had the Visual Studio Code which was updated to 2020 and I deleted it and installed Visual Studio 2019. It no longer gave me the "vcvarsall.bat" error and is currently running tests to ensure proper raven installation. 

Thank you again for your time to help me troubleshoot this!
